src/OwnedConfigHelper.php

```diff
@@ -91,9 +91,63 @@ public static function flattenConfig(array $config): array {
    *
    * @return array
    *   The resulting array.
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
    */
   public static function removeThirdPartySettings(array $config, array $ignored_keys = []) {
```

Comment:
Shall we maybe open a follow up to simplify this and / or write some unit tests for this specific method?

Reply:
Yupp!

Reply:
Follow up created here https://webgate.ec.europa.eu/CITnet/jira/browse/OPENEUROPA-1380